hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/containerlog/parser/ContainerDatanodeDatabase.java

```diff
@@ -233,5 +237,173 @@ private void dropTable(String tableName, Statement stmt) throws SQLException {
+        case "OPEN": openReplicas.add(datanodeId); break;
+        case "QUASI_CLOSED": quasiclosedReplicas.add(datanodeId); break;
+        case "DELETED": deletedReplicas.add(datanodeId); bcsids.add(bcsid); break;
+        default:
+          break;
+        }
+      }
+      
+      int unhealthyCount = unhealthyReplicas.size();
+      int replicaCount = datanodeIds.size();
+      int expectedReplicationFactor = 3;
+
+      if (!deletedReplicas.isEmpty() && !closedReplicas.isEmpty() && bcsids.size() > 1) {
+        System.out.println("Container " + containerID + " has MISMATCHED REPLICATION.");
+      } else if (unhealthyCount == replicaCount) {
+        System.out.println("Container " + containerID + " is UNHEALTHY across all datanodes.");
+      } else if (unhealthyCount >= 2 && closedReplicas.size() == replicaCount - unhealthyCount) {
+        System.out.println("Container " + containerID + " is both UNHEALTHY and UNDER-REPLICATED.");
+      } else if (unhealthyCount == 1 && closedReplicas.size() == replicaCount - unhealthyCount) {
+        System.out.println("Container " + containerID + " is UNDER-REPLICATED.");
+      } else if ((!openReplicas.isEmpty() && openReplicas.size() < 3) &&
+          (closedReplicas.size() == replicaCount - openReplicas.size() ||
+              unhealthyCount == replicaCount - openReplicas.size())) {
+        System.out.println("Container " + containerID + " is OPEN_UNHEALTHY.");
+      } else if (quasiclosedReplicas.size() >= 3) {
+        System.out.println("Container " + containerID + " is QUASI_CLOSED_STUCK.");
+      } else if (replicaCount - deletedReplicas.size() < expectedReplicationFactor) {
+        System.out.println("Container " + containerID + " is UNDER-REPLICATED.");
+      } else if (replicaCount - deletedReplicas.size() > expectedReplicationFactor) {
+        System.out.println("Container " + containerID + " is OVER-REPLICATED.");
+      } else {
+        System.out.println("Container " + containerID + " has enough replicas.");
+      }
+    } catch (SQLException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Checks whether the specified container has multiple "OPEN" states.
+   * If multiple "OPEN" states are found, it returns {@code true}.
+   *
+   * @param entries The list of {@link DatanodeContainerInfo} entries to check for multiple "OPEN" states.
+   * @return {@code true} if multiple "OPEN" states are found, {@code false} otherwise.
+   */
+  
+  private boolean checkForMultipleOpenStates(List<DatanodeContainerInfo> entries) {
+    List<String> firstOpenTimestamps = new ArrayList<>();
+    Set<String> firstOpenDatanodes = new HashSet<>();
+    boolean issueFound = false;
+
+    for (DatanodeContainerInfo entry : entries) {
+      if ("OPEN".equalsIgnoreCase(entry.getState())) {
+        if (firstOpenTimestamps.size() < 3 && !firstOpenDatanodes.contains(entry.getDatanodeId())) {
```

Comment:
I'm assuming you're using 3 here, as the default replication factor is 3. I'm not sure if we can fetch the replication factor dynamically, but let's use a variable instead of hard-coding.